src/promptflow-core/promptflow/contracts/flow.py

```diff
@@ -754,15 +761,15 @@ def from_yaml(cls, flow_file: Path, working_dir=None, name=None) -> "Flow":
             flow_dag = load_yaml(fin)
         # Name priority: name from payload > name from yaml content > working_dir.stem
         # For portal created flow, there is a meaningless predefined name in yaml, use name from payload to override it.
-        if name is None:
-            name = flow_dag.get("name", _sanitize_python_variable_name(working_dir.stem))
-        flow_dag["name"] = name
-        return Flow._from_dict(flow_dag=flow_dag, working_dir=working_dir)
+        return Flow._from_dict(flow_dag=flow_dag, working_dir=working_dir, name=name)
 
     @classmethod
-    def _from_dict(cls, flow_dag: dict, working_dir: Path) -> "Flow":
+    def _from_dict(cls, flow_dag: dict, working_dir: Path, name=None) -> "Flow":
         """Load flow from dict."""
         cls._update_working_dir(working_dir)
+        if name is None:
```

Comment:
When will we use name?

Reply:
Someone used to use it before, only modified the code structure.
![image](https://github.com/microsoft/promptflow/assets/23182548/7801e8c2-65f3-4a50-bf81-f04f53c6212e)